adapter/noop2/noop2.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package noop2
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/golang/protobuf/proto"
+
+	adapter "istio.io/mixer/pkg/adapter"
+	adapter_cnfg "istio.io/mixer/pkg/adapter/config"
+	sample_report "istio.io/mixer/pkg/templates/sample/report"
+)
+
+type (
+	noop2Handler        struct{}
+	noop2HandlerBuilder struct{}
+)
+
+///////////////// Configuration time Methods ///////////////
+
+func (noop2HandlerBuilder) DefaultConfig() proto.Message { return &types.Empty{} }
```

Comment:
Maybe DefaultConfig should be a field on BuilderInfo instead of a function? The value returned is not builder-specific, it's the default config for the entire adapter.

In fact, maybe ValidateConfig should also just be a function pointer on BuilderInfo? Again, it's not really a builder-related function. It's completely stateless and Mixer just needs to get a pointer to it.